src/rulesets/Base/array.jl

```diff
@@ -37,6 +37,12 @@ function rrule(
     return Base.vect(X...), vect_pullback
 end
 
+# This rule isn't always correct, but Zygote and Diffractor fall over completely without it.
+function rrule(::typeof(Base.vect), X...)
+    vect_pullback(ȳ) = (NoTangent(), ȳ...)
```

Comment:
I think this will make `@code_warntype` happier:
```suggestion
function rrule(::typeof(Base.vect), X::Vararg{Any,N}) where {N}
    vect_pullback(ȳ) = (NoTangent(), ntuple(n -> ȳ[n], N)...)
```

Reply:
We should think about making some helpers around this